src/assets/src/components/common.tsx

```diff
@@ -326,19 +349,35 @@ export const SingleInputFormShowRemaining: React.FC<SingleInputFormShowRemaining
                 />
             </Form.Group>
             <div className="remaining-controls-group">
-                <span className={isInvalid ? 'text-danger' : undefined}>{feedbackText}</span>
-                <Button bsPrefix={buttonClass} type='submit' disabled={props.disabled || isInvalid}>{props.children}</Button>
+                {feedbackEl}
+                <Button bsPrefix={buttonClass} type='submit' disabled={props.disabled || isInvalid}>
+                    {props.children}
+                </Button>
             </div>
         </Form>
     );
 }
 
+export enum EditToggleFieldType {
+    inputGroup,
+    textArea
+}
 
-interface ShowRemainingFieldProps extends EditToggleFieldProps {
-    maxLength: number;
+interface EditToggleFieldProps {
+    fieldType: EditToggleFieldType;
+    text: string;
+    placeholder: string;
+    disabled: boolean;
+    buttonType: BootstrapButtonTypes;
+    id: string;
+    onSubmit: (value: string) => void;
+    initialState: boolean;
+    validateLength: boolean;
+    maxLength?: number;
+    allowBlank?: boolean;
```

Comment:
I suppose I could tweak `allowBlank` to be `minLength`, maybe set us up for uniqname validation.

Reply:
I think the validation refactor will be a better way to make this more robust.